tools/test/iree-run-module-inputs.mlir

```diff
@@ -0,0 +1,74 @@
+// Passing no inputs is okay.
+
+// RUN: (iree-compile --iree-hal-target-backends=vmvx %s | \
+// RUN:  iree-run-module --device=local-sync --module=- --function=no_input) | \
+// RUN: FileCheck --check-prefix=NO-INPUT %s
+// NO-INPUT-LABEL: EXEC @no_input
+func.func @no_input() {
+  return
+}
+
+// -----
+
+// Scalars use the form `--input=value`. Type (float/int) should be omitted.
+//   * Type conversions between bit depths (e.g. i8 -> i32) may occur!
+
+// RUN: (iree-compile --iree-hal-target-backends=vmvx %s | \
+// RUN:  iree-run-module --device=local-sync --module=- --function=scalars \
+// RUN:  --input=1 --input=5 --input=1234 --input=-3.14) | \
+// RUN: FileCheck --check-prefix=INPUT-SCALARS %s
+// INPUT-SCALARS-LABEL: EXEC @scalars
+func.func @scalars(%arg0: i1, %arg1: i8, %arg2 : i32, %arg3 : f32) -> (i1, i8, i32, f32) {
+  // INPUT-SCALARS: result[0]: i32=1
+  // INPUT-SCALARS: result[1]: i32=5
+  // INPUT-SCALARS: result[2]: i32=1234
+  // INPUT-SCALARS: result[3]: f32=-3.14
+  return %arg0, %arg1, %arg2, %arg3 : i1, i8, i32, f32
+}
```

Comment:
(So I can have an accurate comment here)

Why are the `i1` and `i8` types turned into `i32` in the output? Passing `--input=2048` yields `result[0]: i32=2048` (neither i1 nor i8 can fit 2048 :P)

Reply:
there's no i1 or i8 types in the vm - you get vm types back, not the original mlir types

Reply:
Ahhhh, right. Is there some promotion happening in the module itself though? I was hoping that `--input=2048` would at least return me a value bounded by the limits of the declared types (either i1 or i8).

I could try with a dispatch - these tests are just passthroughs (inputs -> outputs).

Reply:
there's only vm types, nothing else - so when you say `--input=2048` we parse it as an integer because we can only tell if something is an integer. in fact, today I don't even think i64 is fully supported (we could by checking the vm calling convention but don't - that'd be a good thing to fix). binding layers can add additional validation/verification/type semantics in ways that make sense for them and their type systems but these are just command line developer tools splatting bytes into memory and not meant to be something users are relying on full i3 or f8/f16 or whatever semantics the original program may be trying to model.

Reply:
(think of what we're doing here like va_args in C - there's no way to pass an int8_t as a vararg, just int32_t and int64_t, and it's up to the other side to interpret it in various ways if it wants to)